### PR TITLE
Make Optional People Questions Required if 'yes' is Selected

### PR DESCRIPTION
**Pull request by ascii-dev**
## Description

This PR ensures that optional questions are validated if yes is selected in `PEOPLE IN THE NEWS` tab.

Fixes #174639229

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: forms/static/forms/admin/dependency_fields.js [hunk 1]
```diff
@@ -9,17 +9,40 @@
             $('.field-victim_of').hide();
             $('.field-survivor_of').hide();
         }
+
```
Comment (kilemensi) [inline forms/static/forms/admin/dependency_fields.js]:
```suggestion
```

File: forms/static/forms/admin/dependency_fields.js [hunk 2]
```diff
@@ -9,17 +9,40 @@
...
 
         if((newspaper_person > 0) || (radio_person > 0) || (tv_person > 0) || (internent_person > 0)){
+            addValue('id_newspaperperson_set-0-victim_of');
+            addValue('id_newspaperperson_set-0-survivor_of');
+            addValue('id_radioperson_set-0-victim_of');
+            addValue('id_radioperson_set-0-survivor_of');
+            addValue('id_televisionperson_set-0-victim_of');
+            addValue('id_televisionperson_set-0-survivor_of');
+            addValue('id_internetnewsperson_set-0-victim_of');
+            addValue('id_internetnewsperson_set-0-survivor_of');
```
Comment (kilemensi) [inline forms/static/forms/admin/dependency_fields.js:33]: In a single sheet we can only have one of `newspaper_person`, `radio_person`, etc., right? So why not just handle that e.g.
```js
if((newspaper_person > 0) || (radio_person > 0) || (tv_person > 0) || (internent_person > 0)){
  if (newspaper_person > 0) {
     addValue('id_newspaperperson_set-0-victim_of');
     addValue('id_newspaperperson_set-0-survivor_of');
  }

  ....

```

**Review by kilemensi — APPROVED** [review]
:rocket: @ascii-dev 

-------------------------

Is this going to play nice with https://github.com/CodeForAfrica/gmmp/pull/74?

**Comment by kilemensi** [comment]
Oh... Forgot one issue @ascii-dev.

After validating that the optional questions are required, the **tab** itself doesn't show that that specific panel needs attention i.e. doesn't get the red underline:
Peek 2020-09-08 09-03

This means people won't know to check for errors there.

**Comment by ascii-dev** [comment]
@kilemensi I don't think it clashes with #74. Had a discussion with @esirK and basically what his PR does is to make sure that optional fields don't get validated and are hidden from the form when their triggers are not selected while mine makes sure that when the trigger for this specific form field is triggered but the field isn't filled in, it should throw an error.

**Comment by ascii-dev** [comment]
> Oh... Forgot one issue @ascii-dev.
> 
> After validating that the optional questions are required, the **tab** itself doesn't show that that specific panel needs attention i.e. doesn't get the red underline:
> Peek 2020-09-08 09-03
> 
> This means people won't know to check for errors there.

I think this bug has been existing in the codebase even before now. I'll work on checking what the issue might be and see how I might be able to get it fixed...

File: forms/static/forms/admin/dependency_fields.js (hunk 2 above)
Comment (ascii-dev) [reply forms/static/forms/admin/dependency_fields.js:33]: Oh... The `if` statement was already existing to toggle the fields, I just added my code to it. @kilemensi

**Pull request merged**